build-release-tools/application/releaseNoteAuto.py

```diff
@@ -0,0 +1,160 @@
+import traceback
+import sys
+import requests
+import json
+import configparser
+def release_note_args_parser(args):
+    """
+    This is a function to automatic generate the release notes on conflunence
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--release-version",
+                          help = "This arguments present the version of release note",
+                          action = "store")
+                          
+                          
+    parser.add_argument("--release-startDate",
+                          help = "This arguments present the start date of release note",
+                          action = "store")
+                          
+                          
+    parser.add_argument("--release-endDate",
+                          help = "This arguments present the end date of release note",
+                          action = "store")    
+                          
+    parser.add_argument("--userName",
+                          help = "This arguments present conflunence user name",
+                          action = "store")    
+
+    parser.add_argument("--password",
+                          help = "This arguments present conflunence password",
+                          action = "store")                                  
+    
+    parsed_args = parser.parse_args(args)
+    return parsed_args
+    
+def date_convert(args):
+    """
+    This is a function to convert date format to year-month-day
+    """
+    try:
+         dt = parse(args)
+         return dt.strftime('%Y-%m-%d')  
+         
+    except Exception, e:
+    
+        raise ValueError(e)
+        
+def get_page_content( url_base, page, space, user, pwd)    :
+    """
+    This is a function to get the data from a url
+    """
+    template_url = url_base +  "?title={0}&spaceKey={1}&expand=body.storage".format(page, space)
+    response = requests.get(template_url,  auth=(user, pwd))
+    data = response.json()
+    return data
+
+def get_page_change_version( url_base, page, space, user, pwd)    :
+    """
+    This is a function to get the page's latest version 
+    """
+    template_url = url_base + "?title={0}&spaceKey={1}&expand=version".format(page, space)
+    response = requests.get(template_url,  auth=(user, pwd))
+    data = response.json()
+    return data["results"][0]["version"]["number"]
+
+def create_new_page(configs, user_name,passwd,start_date,end_date,version)   :
+    """
+    This is a function to create a new page for new release
+    """
+    data = get_page_content( configs["rest_url_base"], configs["template_page"], configs["space_key"], user_name, passwd )
+    
+    value = data["results"][0]["body"]["storage"]["value"]
+    value = value.replace( configs["temp_start_date"], start_date)
+    value = value.replace( configs["temp_end_date"],end_date)    
+        
+    header = {'Content-type': 'application/json'}
+    parent_id = data["results"][0]["id"]
```

Comment:
is parent_id the id of template page itself? it should be another page whose page name defined in config